library/src/amd_detail/hipsparse.cpp

```diff
@@ -14617,60 +14824,117 @@ hipsparseStatus_t hipsparseSpGEMM_copy(hipsparseHandle_t          handle,
+
+    void* device_one = (static_cast<char*>(spgemmDescr->externalBuffer2) + byteOffset2);
 
-    const void* alpha_ptr = hipsparse::spgemm_get_ptr(mode, computeType, alpha);
-    const void* beta_ptr  = hipsparse::spgemm_get_ptr(mode, computeType, beta);
-
-    // cuSPARSE API does not carry over the temporary storage buffer, therefore
-    // we need to allocate additional memory. This will lead to lower performance
-    // and thus we highly recommend to use rocSPARSE API instead!!!
+    // Get pointer mode
+    hipsparsePointerMode_t pointer_mode;
+    RETURN_IF_HIPSPARSE_ERROR(hipsparseGetPointerMode(handle, &pointer_mode));
 
-    // Query for required buffer size
-    size_t bufferSize;
-    RETURN_IF_ROCSPARSE_ERROR(rocsparse_spgemm((rocsparse_handle)handle,
-                                               hipsparse::hipOperationToHCCOperation(opA),
-                                               hipsparse::hipOperationToHCCOperation(opB),
-                                               alpha_ptr,
-                                               (rocsparse_const_spmat_descr)matA,
-                                               (rocsparse_const_spmat_descr)matB,
-                                               beta_ptr,
-                                               (rocsparse_spmat_descr)matC,
-                                               (rocsparse_spmat_descr)matC,
-                                               hipsparse::hipDataTypeToHCCDataType(computeType),
-                                               hipsparse::hipSpGEMMAlgToHCCSpGEMMAlg(alg),
-                                               rocsparse_spgemm_stage_buffer_size,
-                                               &bufferSize,
-                                               nullptr));
+    hipStream_t stream;
+    RETURN_IF_HIPSPARSE_ERROR(hipsparseGetStream(handle, &stream));
 
-    void* buffer;
-    RETURN_IF_HIP_ERROR(hipMalloc(&buffer, bufferSize));
+    void* one = spgemm_get_one_ptr(pointer_mode, stream, computeType, device_one);
 
-    hipsparseStatus_t status = hipsparse::rocSPARSEStatusToHIPStatus(
-        rocsparse_spgemm((rocsparse_handle)handle,
-                         hipsparse::hipOperationToHCCOperation(opA),
-                         hipsparse::hipOperationToHCCOperation(opB),
-                         alpha_ptr,
-                         (rocsparse_const_spmat_descr)matA,
-                         (rocsparse_const_spmat_descr)matB,
-                         beta_ptr,
-                         (rocsparse_spmat_descr)matC,
-                         (rocsparse_spmat_descr)matC,
-                         hipsparse::hipDataTypeToHCCDataType(computeType),
-                         hipsparse::hipSpGEMMAlgToHCCSpGEMMAlg(alg),
-                         rocsparse_spgemm_stage_compute,
-                         &bufferSize,
-                         buffer));
+    if(csrColIndTypeC == HIPSPARSE_INDEX_32I)
```

Comment:
Need to add rocsparse_create_identity_permutation32 and rocsparse_create_identity_permutation64 to rocsparse.